api/src/main/java/org/eclipse/microprofile/jwt/builder/JwtClaimsBuilder.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2016-2017 Contributors to the Eclipse Foundation
+ *
+ *  See the NOTICE file(s) distributed with this work for additional
+ *  information regarding copyright ownership.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  You may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.jwt.builder;
+
+import java.util.Set;
+
+/**
+ * JWT Claims Builder.
+ * 
+ * <p>
+ * JwtClaimsBuilder implementations must set the 'iat' (issued at time), 'exp' (expiration time)
+ * and 'jit' (unique token identifier) claims unless they have already been set.
+ * <p>
+ * Note that JwtClaimsBuilder implementations are not expected to be thread-safe.
+ * 
+ * @see <a href="https://tools.ietf.org/html/rfc7519">RFC7515</a>
+ */
+public interface JwtClaimsBuilder extends JwtSignature {
+
+    /**
+     * Set an issuer 'iss' claim
+     * 
+     * @param issuer the issuer
+     * @return JwtClaimsBuilder
+     */
+    JwtClaimsBuilder issuer(String issuer);
+
+    /**
+     * Set a subject 'sub' claim
+     * 
+     * @param subject the subject
+     * @return JwtClaimsBuilder
+     */
+    JwtClaimsBuilder subject(String subject);
+
+    /**
+     * Set a 'upn' claim
+     * 
+     * @param upn the upn
+     * @return JwtClaimsBuilder
+     */
+    JwtClaimsBuilder upn(String upn);
+
+    /**
+     * Set a preferred user name 'preferred_username' claim
+     * 
+     * @param preferredUserName the preferred user name
+     * @return JwtClaimsBuilder
+     */
+    JwtClaimsBuilder preferredUserName(String preferredUserName);
+
+    /**
+     * Set an issuedAt 'iat' claim
+     * 
+     * @param issuedAt the issuedAt time in seconds
+     * @return JwtClaimsBuilder
+     */
+    JwtClaimsBuilder issuedAt(long issuedAt);
```

Comment:
Is this the actual value of the iat or the 'timeStamp'?

Since iat must be in seconds so roughly `new Date().getTime()/1000`

Add overloaded method for Date, LocalDate, ... ?

Reply:
The same what is done [here](https://github.com/eclipse/microprofile-jwt-auth/blob/master/tck/src/main/java/org/eclipse/microprofile/jwt/tck/util/TokenUtils.java#L134) is assumed in this case.

Reply:
But for me, the long parameter is not clear and unambiguous. You can use something like `now()` but also `new Date().getTime()/1000`.  But `iat` should be in seconds and that is not clear from the method signature.

So I propose to use `Date` and `LocalDate` as that is very clear and removes the knowledge requirement from the developer what the exact format of `iat` and `exp` are.